cpp/src/arrow/compute/api_scalar.h

```diff
@@ -216,6 +216,19 @@ struct ARROW_EXPORT ProjectOptions : public FunctionOptions {
   std::vector<std::shared_ptr<const KeyValueMetadata>> field_metadata;
 };
 
+struct ARROW_EXPORT DayOfWeekOptions : public FunctionOptions {
+  explicit DayOfWeekOptions(bool one_based_numbering, uint32_t week_start)
+      : one_based_numbering(one_based_numbering), week_start(week_start) {}
```

Comment:
Can you validate the `week_start` here? (to check that it is in the correct range, and eg a user is not passing 0. Since the numbering here is different as the default output of DayOfWeek, that seems like an easy mistake)

Reply:
I've added this:
```C++
if (week_start < 1 || 7 < week_start) {
  week_start = 1;
}
```

However raising would be better I suppose?
```C++
throw std::invalid_argument("week_start must follow ISO convention (Monday=1, Sunday=7). Got week_start=" + std::to_string(week_start));
```

Reply:
I suppose the Option construction cannot raise an error, since it doesn't return a Result or Status. But we should certainly not silently truncate the value passed by the user. 

Is there another place where this can be validated in the C++ kernel machinery? For example in the kernel's `Exec` ?

Reply:
Moved the check to `Exec` and added some tests.